tests/general/MementoBearTest.py

```diff
@@ -0,0 +1,306 @@
+    else:
+        m.head(
+            'http://timetravel.mementoweb.org/timegate/%s' % url,
+            status_code=404)
+
+
+def generate_redirects(m, url, redirect_length=1):
+    """
+    Generate redirects path for the given url.
+
+    :param m:               A `request_mock.Mocker()` instance.
+    :param url:             The ``url`` to be redirected.
+    :param redirect_length: How many redirects will be generated.
+    """
+    m.head(
+        url,
+        headers={'Location': '%s/1' % url},
+        status_code=302)
+
+    if redirect_length > 1:
+        for i in range(redirect_length - 1):
+            m.head(
+                '%s/%s' % (url, str(i + 1)),
+                headers={'Location': '%s/%s' % (url, str(i + 2))},
+                status_code=302
+            )
+
+        m.head('%s/%s' % (url, str(redirect_length)),
+               status_code=200)
+    else:
+        m.head('%s/1' % url, status_code=200)
+
+
+class MementoBearTest(LocalBearTestHelper):
+    """
+    The tests are mocked (don't actually connect to internet) and
+    return the int conversion of the last three chars of
+    the URL as status code.
+
+    Check ``custom matcher`` for more info on implementation.
+    """
+
+    def assertResultCount(self, test_file, expected_num):
+        MementoBear.check_prerequisites = lambda *args: True
+        uut = MementoBear(self.section, Queue())
+        for line, num in zip(test_file, expected_num):
+            outputs = list(uut.run('testline', [line]))
+            self.assertEqual(num, len(outputs))
+
+    def setUp(self):
+        self.section = Section('')
+        MementoBear.check_prerequisites = lambda *args: True
+        self.uut = MementoBear(self.section, Queue())
+
+    def test_run(self):
+        with requests_mock.Mocker() as m:
+            m.add_matcher(custom_matcher)
+
+            # Register archived links
+            memento_archive_status_mock(m, 'https://www.google.com')
+            memento_archive_status_mock(m, 'https://www.facebook.com')
+            memento_archive_status_mock(
+                m, 'https://github.com/mementoweb/py-memento-client/')
+
+            # Register not archived links
+            memento_archive_status_mock(
+                m, 'https://www.facebook.com/coala', False)
+
+            valid_file = """
+            # Archived links should not yields error
+            https://www.google.com
+            https://www.facebook.com
+            https://github.com/mementoweb/py-memento-client/
+
+            http://httpbin.org/status/401  # Unauthorized
+
+            # Templated URLs can't be checked
+            "http://httpbin.org/{status}/404".format(...)
+            "http://httpbin.org/$status/404"
+
+            # Not a link
+            http://not a link dot com
+            http://www.%s.com
+            http://www.%d.com
+            http://www.%f.com
+
+            # Example.com URLs should be ignored
+            http://sub.example.com/404
```

Comment:
This bear only needs one 404 link.  https://github.com/coala/coala-bears/issues/1726 for fixing the redundancy in InvalidLinkBear.